bundles/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ModbusBinding.java

```diff
@@ -113,23 +112,24 @@ protected void internalUpdateItem(String slaveName, InputRegister[] registers, S
                 continue;
             }
 
-            String slaveValueType = modbusSlaves.get(slaveName).getValueType();
-            double rawDataMultiplier = modbusSlaves.get(slaveName).getRawDataMultiplier();
+            ModbusSlave slave = modbusSlaves.get(slaveName);
+            String slaveValueType = slave.getValueType();
+            double rawDataMultiplier = slave.getRawDataMultiplier();
 
             State newState = extractStateFromRegisters(registers, config.readIndex, slaveValueType);
             /* receive data manipulation */
             State newStateBoolean = provider.getConfig(itemName)
                     .translateBoolean2State(!newState.equals(DecimalType.ZERO));
             if (!UnDefType.UNDEF.equals(newStateBoolean)) {
                 newState = newStateBoolean;
-            } else if ((rawDataMultiplier != 1) && (config.getItem() instanceof NumberItem)) {
+            } else if ((rawDataMultiplier != 1) && (config.getState() instanceof DecimalType)) {
```

Comment:
Is there any chance `config.getState()` could return `null` or an instance of `UnDefType`, causing this `if` block to be skipped, and thus `newState` not getting set?  Or at least, is there any chance of different behavior than when the item was being checked before this PR?  Might be OK; just wanting to confirm.